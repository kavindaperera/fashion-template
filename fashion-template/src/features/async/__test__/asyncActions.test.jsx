import configureStore from 'redux-mock-store';
// Actions to be tested
import * as selectActions from '../asyncActions';

const mockStore = configureStore();
const store = mockStore();


describe('asyncActionStart', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
        {
          'type': 'ASYNC_ACTION_START',
        },
      ];
      store.dispatch(selectActions.asyncActionStart());
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

describe('asyncActionFinish', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
        {
          'type': 'ASYNC_ACTION_FINISH',
        },
      ];
      store.dispatch(selectActions.asyncActionFinish());
      expect(store.getActions()).toEqual(expectedActions);
    });
  });

describe('asyncActionError', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      const expectedActions = [
        {
          'type': 'ASYNC_ACTION_ERROR',
        },
      ];
      store.dispatch(selectActions.asyncActionError());
      expect(store.getActions()).toEqual(expectedActions);
    });
  }); 


 /*SnapShot Testing*/ 

describe('asyncActionStart', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      store.dispatch(selectActions.asyncActionStart());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

describe('asyncActionFinish', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      store.dispatch(selectActions.asyncActionFinish());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

describe('asyncActionError', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });
    test('Dispatches the correct action and payload', () => {
      store.dispatch(selectActions.asyncActionError());
      expect(store.getActions()).toMatchSnapshot();
    });
  });
