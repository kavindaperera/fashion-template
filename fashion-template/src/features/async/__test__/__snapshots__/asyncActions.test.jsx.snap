// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asyncActionError Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "ASYNC_ACTION_ERROR",
  },
]
`;

exports[`asyncActionFinish Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "ASYNC_ACTION_FINISH",
  },
]
`;

exports[`asyncActionStart Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "ASYNC_ACTION_START",
  },
]
`;
